### PR TITLE
The global .gitignore ignores DLL, this is a problem with plugins in …

### PR DESCRIPTION
**Pull request by allanolivei**
…Unity 3D

**Reasons for making this change:**

_TODO_

**Links to documentation supporting these rule changes:**

_TODO_

If this is a new template:

 - **Link to application or project’s homepage**: _TODO_

### CODE REVIEW

File: Unity.gitignore [hunk 1]
```diff
@@ -39,3 +39,5 @@ sysinfo.txt
 # Crashlytics generated file
 Assets/StreamingAssets/crashlytics-build.properties
 
+#Do not ignore plugins
+!/**/*.dll
```
Comment (shiftkey) [inline Unity.gitignore]: This came up in #1603 and after a bit of discussion I think I was concerned about this being recursively applied to all paths - can this rule be made more specific?

Comment (allanolivei) [reply Unity.gitignore]:
```suggestion
!/Assets/Plugins/**/*.dll
```

Comment (shiftkey) [reply Unity.gitignore]: I guess? I'd need some input from people who are more familiar with Unity at this point.

cc @ksmit799 @theDrake @1nfinite9 @owenblacker @FutureClemens @kilian442 @mariuz for feedback (peeps who have made changes to the template over the last year)

Comment (FutureClemens) [reply Unity.gitignore]: DLLs in a Unity project are allowed to be placed within any "Plugins" directory or "Plugins" sub-directory. Also, the "Plugins" directory itself does not need to be placed directly in the Assets folder and can be placed in any directory. And there might be multiple plugins directories in your project.

Therefore, valid places are:
Assets\[0-n Directories]\Plugins\[0-n Directories]\plugin.dll

Examples from existing plugins, that are working in my projects (Unity 2017 & 2018):
Assets\Plugins\x86_64\plugin.dll
Assets\Plugins\WSA\PhoneSDK81\x86\plugin.dll
Assets\MyTests\Plugins\x86_64\Test.dll

I am not sure if a line like this makes sense at all for a unity gitignore file, but if you do it correctly, it at least does not do any harm.

Comment (allanolivei) [reply Unity.gitignore]:
```suggestion
!/Assets/**/Plugins/**/*.dll
```

capture

Thanks @FutureClemens.

Comment (ksmit799) [reply Unity.gitignore]: I'm not sure if this makes sense within the scope of the global Unity gitignore, seeing as the current Unity gitignore doesn't ignore any dll files by itself. The fact that an external gitignore is impacting on Unity's makes this more of an edge case then something that should be applied globally.

**Comment by shiftkey** [comment]
Closing this out due to inactivity and outstanding review feedback.

If someone else thinks these changes would still add value to the template, please open a fresh PR.

**Pull request closed without merging**